src/module-elasticsuite-catalog/Model/ResourceModel/Product/Fulltext/Collection.php

```diff
@@ -615,6 +627,82 @@ protected function _afterLoad()
         return parent::_afterLoad();
     }
 
+    /**
+     * Lazily get the QueryFactory instance.
+     *
+     * @return QueryFactory
+     */
+    protected function getQueryFactory()
+    {
+        if ($this->queryFactory === null) {
+            $this->queryFactory = \Magento\Framework\App\ObjectManager::getInstance()
+                ->get(QueryFactory::class);
+        }
+
+        return $this->queryFactory;
+    }
+
+
+    /**
+     * Prepares min and max price using Elasticsearch metric aggregation.
+     * Ensures compatibility with Magento Core's getMinPrice()/getMaxPrice().
+     *
+     * @return $this
+     */
+    protected function _prepareStatisticsData()
+    {
+        $storeId = $this->getStoreId();
+        $requestName = $this->searchRequestName;
+        $customerGroupId = (int) ($this->_productLimitationFilters['customer_group_id'] ?? CustomerGroup::NOT_LOGGED_IN_ID);
+        $aggregationName = 'price_stats';
```

Comment:
Could you rename it _collection_price_stats ?
It's just to be a bit further in name from the "price.price.stats" legacy one that can appear with the adaptative slider enabled.